examples/RoleAssignment/roleAssignment.bicep

```diff
@@ -0,0 +1,14 @@
+targetScope = 'subscription'
+
+@description('Service Principal Object Id')
+param principalId string
+
+var readerRole = subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
```

Comment:
Is this a hard-coded GUID, or is this a "known" GUID?

Reply:
This is a known GUID. All of the roles in Azure have their own corresponding GUID, and the azure documentation for assigning roles does follow this pattern of using a hard coded GUID. [https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles]( https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles)